frontend/app/src/App.tsx

```diff
@@ -717,6 +718,16 @@ export class App extends PureComponent<Props, State> {
           this.handleLogo(logo, msgProto.metadata as ForwardMsgMetadata),
         navigation: (navigation: Navigation) =>
           this.handleNavigation(navigation),
+        authRedirect: (authRedirect: AuthRedirect) => {
+          if (isInChildFrame()) {
+            this.hostCommunicationMgr.sendMessageToSameOriginHost({
+              type: "REDIRECT_TO_URL",
+              url: authRedirect.url,
+            })
```

Comment:
Sorry, forgot to respond here :) 

yes, in this case instead of standard `sendMessageToHost` we use `sendMessageToSameOriginHost` to ensure that just by embedding popular streamlit app into your iframe you will be not able to intercept authentication request. 

We send the message to parent host, so other Streamlit hosting providers would be able to implement their changes to make native streamlit authentication work on their platform.